### PR TITLE
Get BusType differently due to MSDN API description from Microsoft

### PR DESCRIPTION
**Pull request by MHumm**
Implemented the suggestion from Mantis issue 6538, but fixing the mistakes that code contained (wrong parameter names and wrong method called)
[http://issuetracker.delphi-jedi.org/view.php?id=6538](url)

### CODE REVIEW

File: jvcl/run/JvHidControllerClass.pas [hunk 1]
```diff
@@ -755,6 +757,21 @@ function TJvHidPnPInfo.GetRegistryPropertyDWord(PnPHandle: HDEVINFO;
     RegDataType, PBYTE(@Result), SizeOf(Result), BytesReturned);
 end;
 
+function TJvHidPnPInfo.GetRegistryPropertyGuid(APnPHandle: HDEVINFO;
+  const ADevData: TSPDevInfoData; Prop: DWORD): TGuid;
+const
+  GUID_NULL: TGUID = '{00000000-0000-0000-0000-000000000000}';
```
Comment (obones) [inline jvcl/run/JvHidControllerClass.pas]: The RTL already has `EmptyGUID` defined, no need to redefine it. And if it is missing in older versions of Delphi, this constant my has well be defined in `JvJCLUtils` for instance

**Comment by MHumm** [comment]
Ok, we can change this to EmptyGUID, but how to find out since when this constant is available in the RTL?
I did a quick check in JvJCLUtils, but I didn't find such a constant there. So your idea is to find out in which Delphi version this EmptyGUID constant was introduced and if that was later as D6, we implement our own copy of it in JvJCLUtils and pull in the right one via conditional define?

**Comment by MHumm** [comment]
Some research found this one: GUID_NULL. It is in System.Types but according to this conversation from 2004 it was in system.pas itsself in former times:
http://www.delphigroups.info/2/a0/507657.html
Unfortunately is D6 older than 2004.

**Comment by MHumm** [comment]
Ok, I asked about that one here (in German):
[https://www.delphipraxis.net/200518-wann-wurde-guid_null-eingefuehrt.html#post1431147](url)

**Comment by obones** [comment]
I have a VM at work with all versions of Delphi from D6 to Rio installed, I'll check on Monday

**Comment by MHumm** [comment]
No need to. Delphipraxis folks already confirmed it is available in D6 already. Will fix it later and resubmit.

**Comment by MHumm** [comment]
Closed as replaced  #by106

**Comment by MHumm** [comment]
Closed as replaced #by106

**Pull request closed without merging**